Sources/CasePaths/EnumReflection.swift

```diff
@@ -69,22 +71,37 @@ public func extract<Root, Value>(case embed: @escaping (Value) -> Root, from roo
 public func extract<Root, Value>(_ embed: @escaping (Value) -> Root) -> (Root) -> (Value?) {
   var cachedTag: UInt32?
   return { root in
-    guard let rootTag = enumTag(root) else { return nil }
-    if let cachedTag = cachedTag, cachedTag != rootTag { return nil }
+    guard let metadata = EnumMetadata(Root.self) else { return nil }
+    let rootTag = metadata.tag(of: root)
+
+    if let cachedTag = cachedTag {
+      guard cachedTag == rootTag else { return nil }
+      if metadata.typeDescriptor.fieldDescriptor.field(atIndex: rootTag).isIndirectCase {
+        return .some(metadata.indirectAssociatedValue(of: root, as: Value.self))
+      } else {
+        return .some(metadata.directAssociatedValue(of: root, as: Value.self))
+      }
+    }
+
     guard
       let value = (Mirror(reflecting: root).children.first?.value)
         .flatMap({ $0 as? Value ?? Mirror(reflecting: $0).children.first?.value as? Value })
     else {
       #if compiler(<5.2)
         // https://bugs.swift.org/browse/SR-12044
-        if MemoryLayout<Value>.size == 0, !isUninhabitedEnum(Value.self) {
+        if
+          MemoryLayout<Value>.size == 0,
+          let valueMetadata = EnumMetadata(Value.self),
+          valueMetadata.kind == .enumeration,
+          valueMetadata.typeDescriptor.isInhabited
+        {
           return unsafeBitCast((), to: Value.self)
         }
       #endif
       return nil
     }
     if rootTag == cachedTag { return value }
```

Comment:
Not sure if it was possible to hit this code path before, but definitely looks impossible now :smile:
```suggestion
```

Reply:
Yeah, that can be removed.